### PR TITLE
Fix two use-after-free bugs

### PR DESCRIPTION
**Pull request by MaxKellermann**
Plus a little bit of code cleanup.

### CODE REVIEW

**Comment by ksooo** [comment]
Thanks Max, much appreciated.

File: xbmc/epg/GUIEPGGridContainer.h [hunk 1]
```diff
@@ -41,6 +38,8 @@ namespace EPG
 
   class CGUIEPGGridContainer : public IGUIContainer
   {
+    static constexpr size_t MAXBLOCKS = (33 * 24 * 60 / 5); //! 33 days of 5 minute blocks (31 days for upcoming data + 1 day for past data + 1 day for fillers)
```
Comment (Paxxi) [inline xbmc/epg/GUIEPGGridContainer.h]: Constexpr isn't supported in vs2013 so this will break the build

Comment (MaxKellermann) [inline xbmc/epg/GUIEPGGridContainer.h]: *sigh* VS is annoying. I wish you'd drop VS support and switch to mingw32.

**Comment by MaxKellermann** [comment]
Dropped the constexpr patch because VS can't compile it apparently.

**Comment by ksooo** [comment]
> What I won't do is change this commit to copy everything. I consider that a waste of my time.

 Hmm. This one is tricky as the copy ctor a) is not working correctly today, b) is not working correctly after your change and c) it is actually not used which in fact means it doesn't matter how it looks like today. If you don't have the time to fix it properly, I think it is best to drop your commit for now. Again, as the copy ctor is currently never called it doesn't really matter whether you fixed the "free after use" problem there. And if somebody actually calls the copy ctor in the future (I doubt that btw) he will learn very fast that it is broken. ;-) Just my 2 ct.

**Comment by MaxKellermann** [comment]
> is not working correctly after your change
Depends on your definition of "correct". CGUIControl::Clone() is undocumented, and we don't how much of a control's volatile state is supposed to be copied.

**Comment by fritsch** [comment]
Just make it private :-)

**Comment by ksooo** [comment]
@MaxKellermann Sorry, it was not my intention to offend you. If I did, sorry. Your PR contained lots of very good changes - "much appreciated" - as I wrote here. Why do you close the whole PR just because of one single commit we need to sort out?

**Pull request closed without merging**